receiver/snmpreceiver/config.go

```diff
@@ -0,0 +1,592 @@
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+// Config Defaults
+const (
+	defaultCollectionInterval = 10 // In seconds
+	defaultEndpoint           = "udp://localhost:161"
+	defaultVersion            = "v2c"
+	defaultCommunity          = "public"
+	defaultSecurityLevel      = "no_auth_no_priv"
+	defaultAuthType           = "MD5"
+	defaultPrivacyType        = "DES"
+)
+
+// Config error messages
+var (
+	errMsgInvalidEndpoint                  = `invalid endpoint '%s': %w`
+	errMsgNoHostInvalidEndpoint            = `invalid endpoint '%s': must be in '[scheme]://[host]:[port]' format`
+	errMsgAttributeConfigNoEnumOIDOrPrefix = `attribute '%s' must contain one of either an enum, oid, or indexed_value_prefix`
+	errMsgResourceAttributeNoOIDOrPrefix   = `resource_attribute '%s' must contain one of either an oid or indexed_value_prefix`
+	errMsgMetricNoUnit                     = `metric '%s' must have a unit`
+	errMsgMetricNoGaugeOrSum               = `metric '%s' must have one of either a guage or sum`
+	errMsgMetricNoOIDs                     = `metric '%s' must have one of either scalar_oids or indexed_oids`
+	errMsgGaugeBadValueType                = `metric '%s' gauge value_type must be either int, float, or bool`
+	errMsgSumBadValueType                  = `metric '%s' sum value_type must be either int, float, or bool`
+	errMsgSumBadAggregation                = `metric '%s' sum aggregation value must be either cumulative or delta`
+	errMsgScalarOIDNoOID                   = `metric '%s' scalar_oid must contain an oid`
+	errMsgScalarAttributeNoName            = `metric '%s' scalar_oid attribute must contain a name`
+	errMsgScalarAttributeBadName           = `metric '%s' scalar_oid attribute name '%s' must match an attribute config`
+	errMsgScalarOIDBadAttribute            = `metric '%s' scalar_oid attribute name '%s' must match attribute config with enum values`
+	errMsgScalarAttributeBadValue          = `metric '%s' scalar_oid attribute '%s' value '%s' must match one of the possible enum values for the attribute config`
+	errMsgColumnOIDNoOID                   = `metric '%s' column_oid must contain an oid`
+	errMsgColumnAttributeNoName            = `metric '%s' column_oid attribute must contain a name`
+	errMsgColumnAttributeBadName           = `metric '%s' column_oid attribute name '%s' must match an attribute config`
+	errMsgColumnAttributeBadValue          = `metric '%s' column_oid attribute '%s' value '%s' must match one of the possible enum values for the attribute config`
+	errMsgColumnResourceAttributeBadName   = `metric '%s' column_oid resource_attribute '%s' must match a resource_attribute config`
+	errMsgColumnIndexedAttributeRequired   = `metric '%s' column_oid must either have a resource_attribute or an indexed_value_prefix/oid attribute`
+)
+
+var (
```

Comment:
nit: Any reason these vars are in a separate block?

Reply:
changed